advanced-list/src/app.js

```diff
@@ -12,32 +13,26 @@ function makeRandomColor() {
   return hexColor;
 }
 
-function intent(DOM, tickerAction$) {
-  const tickerRemove$ = tickerAction$.filter(a => a.type === 'remove');
-  const action$ = xs.merge(
-    tickerRemove$
-  );
-  return action$;
+function insertReducer(tickers) {
+  const color$ = xs.periodic(1000)
+    .map(makeRandomColor)
+    .startWith('#000000');
+
+  return tickers.add(color$);
 }
 
-function model(action$, TickerComponent) {
+function model(tickers) {
   const insertReducer$ = xs.periodic(5000).take(10)
-    .map(id => function insertReducer(oldList) {
-      const color$ = xs.periodic(1000)
-        .map(makeRandomColor)
-        .startWith('#000000');
-      const out = TickerComponent(color$, id);
-      return oldList.concat([{id, DOM: out.DOM, action$: out.action$}]);
-    });
-
-  const removeReducer$ = action$
-    .filter(a => a.type === 'remove')
-    .map(action => function removeReducer(oldList) {
-      return oldList.filter(item => item.id !== action.id);
-    });
-
-  const list$ = xs.merge(insertReducer$, removeReducer$)
-    .fold((oldList, reducer) => reducer(oldList), [])
+    .mapTo(insertReducer);
+
+  const reducer$ = xs.merge(
+    insertReducer$,
+
+    tickers.effect$
```

Comment:
Yeah, I'd be more comfortable with `reducer$` than `effect$`. Effects reminds us of commands that should happen in drivers. Still, I'm not sure should the hard-coded API be `reducer$` or `reducerStream` or `reducers`. In the new DOM Driver we are using `select().elements` instead of `select().observable` or `select().element$` or `select().elementStream`. It's not easy to decide names, but if I'd have to choose, I'd pick `tickers.reducers` just to stay similar to the DOM driver.